### PR TITLE
Selenium testing

### PR DESCRIPTION
**Pull request by pyotrritchie**

### CODE REVIEW

File: src/Service.Host/client/test/e2e/page-objects/base-component.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        this.selector = selector;
+    }
+
+    checkSelectorExist() {
+        if (this.selector === undefined) {
+            throw new TypeError(
+                `Class '${this.constructor.name}' ` +
+                    `extends 'UIComponent' possibly 'Page' Object Class and have to implement abstract property 'selector' ` +
+                    `when 'isDisplayed' or 'waitUntilDisplayed' are used`
```
Comment (atcampbell) [inline src/Service.Host/client/test/e2e/page-objects/base-component.js]: This is maybe a little unclear?

File: src/Service.Host/client/test/e2e/products/productRanges.integration.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        const listItem = element(by.tagName('a'));
+        await waitUntilDisplayed(listItem);
+        return expect(listItem.isPresent()).to.eventually.be.true;
+        // const createProductRangeTitle = element(by.id('qa-product-range-title'));
+        // await waitUntilDisplayed(createProductRangeTitle);
+        // expect(createProductRangeTitle.isPresent()).to.be.true;
+        // const idField = element(by.id('qa-product-range-id'));
+        // expect(idField.isPresent()).to.be.true;
+        // expect(idField.isEnabled()).to.be.false;
+        // browser.get('/products/maint/product-ranges');
```
Comment (atcampbell) [inline src/Service.Host/client/test/e2e/products/productRanges.integration.js]: Comments

File: src/Service.Host/config-example.env [hunk 3]
```diff
@@ -10,3 +10,5 @@ APP_ROOT=http://localhost:61798
 AUTHORITY_URI=https://www-sys.linn.co.uk/auth/
 PROXY_ROOT=http://app.linn.co.uk
 APP_PATH=.\\
+TEST_USER_NAME=guest
+TEST_USER_PWD=guest
```
Comment (atcampbell) [inline src/Service.Host/config-example.env]: Should these maybe be in a test specific file? I feel like these other variables are application related. If not we'd need to remember to add these to the prod sys and env configs in aws.

File: src/Service.Host/client/test/e2e/products/productRanges.integration.js (hunk 2 above)
Comment (pyotrritchie) [reply src/Service.Host/client/test/e2e/products/productRanges.integration.js]: 👍

File: src/Service.Host/config-example.env (hunk 3 above)
Comment (pyotrritchie) [reply src/Service.Host/config-example.env]: Good idea, I'll put them in a specific file.. They will, however, be needed for the build in travis eventually but a specific test file makes sense..

File: src/Service.Host/client/test/e2e/page-objects/base-component.js (hunk 1 above)
Comment (pyotrritchie) [reply src/Service.Host/client/test/e2e/page-objects/base-component.js]: 👍

**Pull request merged**